src/tvheadend/HTSPDemuxer.cpp

```diff
@@ -663,8 +663,13 @@ bool HTSPDemuxer::AddTVHStream(uint32_t idx, const char* type, htsmsg_field_t *f
       return false;
     }
 
-    /* Setting aspect ratio to zero will cause XBMC to handle changes in it */
```

Comment:
So this is no longer the case?

Reply:
At least on Android Kodi does NOT handle it. Did no further investigations, though.

Reply:
Okay. I'm not super sure about this one. Maybe someone who knows more could chip in?

Reply:
> someone who knows more

 Houston, we've got a problem. Who should that be?

Reply:
if you set ar in demuxer, you have to observe ar changes in demuxer too. those can occur mid-stream.
if ar is not set by demuxer, it is supposed to be handled by decoder. Kodi's Android port seems to be incomplete in this regard.

Reply:
Just found this in the htsp spec (https://tvheadend.org/projects/tvheadend/wiki/Htsp)

```
aspect_num         u32   optional   Aspect ratio numerator (Added in version 5) *Can be incorrect and should not be relied upon*
aspect_den         u32   optional   Aspect ratio denonminator (Added in version 5) *Can be incorrect and should not be relied upon*
```

 ... so, it might be incorrect and this info is only set once on subscription start.

=> I will drop this commit from this PR.